### ISSUE TITLE
register_property with hint enum doesn't show list in the editor

### ISSUE THREAD

**Issue by Coldragon** [GodotNativeTools/godot-cpp#359]
NativeScript: 1.1 (branch 1.1)
Godot: 3.1.2

Hi,

When I try to export a property as an enum it doesn't work and in the editor the list is empty

![image](https://user-images.githubusercontent.com/471393/70843233-40579800-1e2f-11ea-8e12-0bca7e5152f7.png)

here what I wrote

```cpp
register_property<MyClass, int>
(
		"open_transition", 
		&MyClass::set_transititon_type,	
		&MyClass::get_transititon_type,	
		Tween::TRANS_BOUNCE,
		GODOT_METHOD_RPC_MODE_DISABLED, 
		GODOT_PROPERTY_USAGE_DEFAULT, 
		GODOT_PROPERTY_HINT_ENUM, 
		"TRANS_LINEAR, TRANS_SINE, TRANS_QUINT, TRANS_QUART, TRANS_QUAD, TRANS_EXPO, TRANS_ELASTIC, TRANS_CUBIC, TRANS_CIRC, TRANS_BOUNCE, TRANS_BACK"
);
```

I tried to write my hint text like in the comment here https://github.com/GodotNativeTools/godot_headers/blob/3.1/nativescript/godot_nativescript.h#L56

Am I doing it wrong ?

Thank's a lot

**Comment by sheepandshepherd** [comment]
I can't figure this one out. My code looks exactly the same but works in the inspector:
```cpp
	register_property<GDExample, int>(
		"speed",
		&GDExample::set_speed,
		&GDExample::get_speed,
		Tween::TRANS_BOUNCE,
		GODOT_METHOD_RPC_MODE_DISABLED,
		GODOT_PROPERTY_USAGE_DEFAULT,
		GODOT_PROPERTY_HINT_ENUM,
		"TRANS_LINEAR, TRANS_SINE, TRANS_QUINT, TRANS_QUART, TRANS_QUAD, TRANS_EXPO, TRANS_ELASTIC, TRANS_CUBIC, TRANS_CIRC, TRANS_BOUNCE, TRANS_BACK"
	);
```

Are you using an old version of godot-cpp maybe? There were many memory management bugs in the past, including some with String that might have caused the hint_string to be freed too early or something... idk.

Can you add and register a `_ready` method that does this? (It won't work in `_init`):
```cpp
Godot::print(Variant(get_property_list()));
```

**Comment by Coldragon** [comment]
I used the 1.1 branch. I will try
Here is my full project if you wanna directly look in it
[TestGDNative.zip](https://github.com/GodotNativeTools/godot-cpp/files/3964107/TestGDNative.zip)

**Comment by Coldragon** [comment]
```

**Comment by sheepandshepherd** [comment]
Yep, `hint_string` is indeed getting deleted somehow. I would suggest switching to the latest master branch - there's been so many memory bugs fixed since then, one of them is probably responsible.

Btw, I'm not sure what the 1.1 branch is actually for - `master` also uses 1.1. I *think* karroffel merged 1.1 into `master` with a single commit after the 1.0 branch was made.

**Comment by Coldragon** [comment]
Ok, with the master it work well thank's
![image](https://user-images.githubusercontent.com/471393/70852486-2e630d00-1ea2-11ea-9c7e-cef12f213dff.png)
